akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala

```diff
@@ -16,33 +21,60 @@ import akka.persistence.PersistentActor
 private[akka] object EventSourcedRememberShards {
   def props(typeName: String): Props =
     Props(new EventSourcedRememberShards(typeName))
+
+  class FromOldCoordinatorState() extends EventAdapter {
+    override def manifest(event: Any): String =
+      ""
+
+    override def toJournal(event: Any): Any =
+      event
+
+    override def fromJournal(event: Any, manifest: String): EventSeq = {
+      event match {
+        case ShardHomeAllocated(shardId, _) =>
+          EventSeq.single(shardId)
+        case _ => EventSeq.empty
+      }
+
+    }
+  }
 }
 
 /**
  * INTERNAL API
+ *
+ * FIXME add snapshotting and then the migration event adapter can be removed
  */
 @InternalApi
-private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor {
+private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor with ActorLogging {
 
-  override val persistenceId: String = s"$typeName-remember-entitites"
+  // Uses the same persistence id as the old persistent coordinator so that the old data can be migrated
+  // without any user action
```

Comment:
I'm trying to understand. What will we support? Rolling update?

Old is `state-store-mode=persistent` and new must be `state-store-mode=ddata` and `remember-entities-store=eventsourced` ?

Can the coordinator by mistake be started again with `state-store-mode=persistent` and there were some new events were already stored by EventSourcedRememberShards, and those can't be understood by the coordinator?

Reply:
The entire thing will require some migration notes in the end, so maybe keep track of those or add to docs so we don't forget something.

Reply:
We won't support downgrade back to `state-store-mode=persistent`. Perhaps we could store an event to mark that we have migrated, first thing after replaying a migration, that the persistent coordinator does not understand or can know about and fail for?

Reply:
It's fine to not support downgrade. Maybe the old implementation can detect the new events and fail?

Reply:
Yes, I think that it would be nice to give a clear error message rather than obscure replay failure.

Reply:
👍 i'll add docs in this PR as well as detecting that uses have gone back